openfisca_core/entities.py

```diff
@@ -40,6 +40,8 @@ def check_role_validity(self, role):
             raise ValueError("{} is not a valid role".format(role))
 
     def get_variable(self, variable_name, check_existence = False):
+        if not self._tax_benefit_system:
+            raise ValueError("No tax_benefit_system set. Please set it using set_tax_benefit_system")
```

Comment:
It happens when you don't explicitly create a simulation object before calling get_variable. I was getting crashes in the top level API because of this (and this is why we create the simulation in the dependencies endpoint). I thought it's best to raise an actual meaningful exception, rather than crash and have to read through the stack trace if someone does this. It should never happen in the live system, but can happen when you're changing the code.